src/github.rs

```diff
@@ -250,6 +250,38 @@ pub struct Issue {
     repository: OnceCell<IssueRepository>,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct PartialIssue {
+    pub number: u64,
+    pub title: String,
+    pub repository: IssueRepository,
+}
+
+impl PartialIssue {
+    pub fn zulip_topic_reference(&self) -> String {
+        let repo = &self.repository;
+        if repo.organization == "rust-lang" {
+            if repo.repository == "rust" {
+                format!("#{}", self.number)
+            } else {
+                format!("{}#{}", repo.repository, self.number)
+            }
+        } else {
+            format!("{}/{}#{}", repo.organization, repo.repository, self.number)
+        }
+    }
+}
+
+impl From<&Issue> for PartialIssue {
```

Comment:
This clones some of the fields. Is this the right trait for that? I couldn’t think of a better one